dev/com.ibm.ws.concurrent.mp.cdi/src/com/ibm/ws/concurrent/mp/cdi/ManagedExecutorProducer.java

```diff
@@ -0,0 +1,53 @@
+/*******************************************************************************
+ * Copyright (c) 2018 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.concurrent.mp.cdi;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import javax.enterprise.inject.Produces;
+import javax.enterprise.inject.spi.InjectionPoint;
+
+import org.eclipse.microprofile.concurrent.ManagedExecutor;
+import org.eclipse.microprofile.concurrent.ManagedExecutorBuilder;
+import org.eclipse.microprofile.concurrent.ManagedExecutorConfig;
+
+import com.ibm.websphere.ras.Tr;
+import com.ibm.websphere.ras.TraceComponent;
+
+public class ManagedExecutorProducer {
+
+    private static final TraceComponent tc = Tr.register(ManagedExecutorProducer.class);
+
+    // A new instance of this class will be constructed each time the producer is
+    // called so we need to keep the config->executor map in a static variable
+    private static final Map<ManagedExecutorConfigContainer, ManagedExecutor> configMap = new ConcurrentHashMap<>();
+
+    @Produces
+    public ManagedExecutor produceManagedExecutor(InjectionPoint point) {
+        ManagedExecutorConfig configAnno = point.getAnnotated().getAnnotation(ManagedExecutorConfig.class);
+        ManagedExecutorConfigContainer config = ManagedExecutorConfigContainer.get(configAnno);
+        ManagedExecutor exec = configMap.computeIfAbsent(config, this::createManagedExecutor);
```

Comment:
I wonder if the createManagedExecutor function & ManagedExecutorBuilderImpl.build is consistent with the intent of computeIfAbsent's requirement that it be "short and simple".  I suppose it might become simpler as the code evolves.

Reply:
See the [javadoc for the computeIfAbsent method](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ConcurrentHashMap.html#computeIfAbsent-K-java.util.function.Function-).  The spec uses that language, although left to interpretation, so it is unclear if our builder constructor and related processing fits the definition.